### PR TITLE
returnCollateral adapter integration tests

### PR DESCRIPTION
**Pull request by graemecode**

### CODE REVIEW

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/collateralized_simple_interest_loan_adapter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        const servicingApi = new ServicingAPI(web3, contractsApi);
+
+        const adapter = new CollateralizedSimpleInterestLoanAdapter(web3, contractsApi);
+
+        const orderApi = new OrderAPI(web3, contractsApi, adaptersApi);
+
+        const signerApi = new SignerAPI(web3, contractsApi);
+
+        const debtKernel = await DebtKernelContract.deployed(web3, TX_DEFAULTS);
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/collateralized_simple_interest_loan_adapter.spec.ts]: Nit: both `debtKernel` and `repaymentRouter` can be loaded from `contractsAPI` without having to specify TX_DEFAULTS or pass in web3.

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { ReturnCollateralRunner } from "./return_collateral_runner";
+
+export { ReturnCollateralRunner };
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/index.ts]: Nit: there's a cleaner way of doing this I learned recently:

`export { ReturnCollateralRunner } from "./return_collateral_runner";`

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        this.termsContract = contractWrappers.termsContract;
+        this.repaymentRouter = contractWrappers.repaymentRouter;
+        this.principalToken = contractWrappers.principalToken;
+        this.collateralToken = contractWrappers.collateralToken;
+
+        this.orderApi = apis.orderApi;
+        this.signerApi = apis.signerApi;
+        this.servicingApi = apis.servicingApi;
+
+        this.adapter = adapter;
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts]: I think we can obviate the need for this `initialize` method -- all of the above contracts can be loaded in the `beforeAll` block using either the APIs or (in the case of the DummyTokens) their contract wrappers.  Please refactor to remove the dependency on running `initialize` -- it's much less brittle to only rely on the constructor as an initializer.

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+            principalAmount.mul(2),
+            { from: debtor },
+        );
+
+        // The creditor grants allowance of the principal token being loaned,
+        // as well as the creditor fee.
+        await this.principalToken.approve.sendTransactionAsync(
+            this.tokenTransferProxy.address,
+            principalAmount.add(this.debtOrder.creditorFee),
+            { from: creditor },
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts]: In the spirit of this being an integration test, these approvals should all happen using the `tokenAPI`

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/scenarios/index.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    collateralTerms: CollateralizedTermsContractParameters;
+    // A function that returns a valid DebtOrder instance.
+    generateDebtOrder: (
+        debtKernel: DebtKernelContract,
+        repaymentRouter: RepaymentRouterContract,
+        principalToken: DummyTokenContract,
+        termsContract: CollateralizedSimpleInterestTermsContractContract,
+        termsParams: string,
+    ) => DebtOrder.Instance;
+    // Typically the underwriter fills the debt order.
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/scenarios/index.ts]: This is not typically the case -- more of an unusual case if anything.  Please fix comment to clarify.

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+        );
+
+        await this.signOrder();
+
+        await this.setBalances(scenario);
+
+        await this.setApprovals(scenario);
+
+        await this.orderApi.fillAsync(this.debtOrder, {
+            from: UNDERWRITER.address,
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts]: Shouldn't this use the `scenario.orderFiller`?

File: __test__/integration/adapters/collateralized_simple_interest_loan_adapter/scenarios/index.ts [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        description: "default description",
+        succeeds: true,
+        termLapsed: true,
+        debtRepaid: true,
+        orderFiller: underwriter.address,
+        collateralTerms: defaultCollateralTerms,
+        simpleTerms: defaultSimpleTerms,
+        collateralWithdrawn: false,
+        givenAgreementId: (agreementId: string) => agreementId,
+        generateDebtOrder: (
```
Comment (nadavhollander) [inline __test__/integration/adapters/collateralized_simple_interest_loan_adapter/scenarios/index.ts]: It doesn't seem like we ever deviate away from this default for `generateDebtOrder` -- why not instead generate this debt order in the Runner?

**Comment by graemecode** [comment]
@nadavhollander I think I've completed those requests.

Note, 2 failing examples here:
```
#returnCollateral
      Successful return of collateral
        ...
        when the debt's term has not yet lapsed and the debt has been paid
          ✕ returns a valid transaction hash (218ms)
          ✕ transfers collateral back to the debtor (34ms)
```

I think this means that the latest collateralizer contract is not deployed.

**Pull request merged**